StrongInject.Generator/Disposal.cs

```diff
@@ -8,5 +8,6 @@ namespace StrongInject.Generator
         internal sealed record DisposalHelpers(string VariableName, bool IsAsync) : Disposal(IsAsync);
         internal sealed record FactoryDisposal(string VariableName, string FactoryName, bool IsAsync) : Disposal(IsAsync);
         internal sealed record DelegateDisposal(string DisposeActionsName, ITypeSymbol DisposeActionsType, bool IsAsync) : Disposal(IsAsync);
+        internal sealed record OnlyDisposeIfInitializationFails(Disposal Disposal) : Disposal(Disposal.IsAsync);
```

Comment:
Why do we need this?

Reply:
There was no existing way for the same operation to cause variable initialization and disposal if initialization fails without also final disposal being handled in the same scope.

I'll unsquash a commit so that you can see the effect this had on the test diff.

Reply:
I see thanks. I'm guessing that if (like delegate) the dependent operations of the Owned weren't added to the main order but kept as a field of OwnedStatement, and generated in a local function, that this wouldn't be necessary.

Basically if you try generating it exactly like delegate I think a lot of aspects would be much simpler.

Reply:
If they aren't added to the main order, then the variable declarations and instantiations don't happen. What could I do instead?

Reply:
They happen the same way delegate does - by generating the code recursively.

Reply:
:+1: I think I started putting the pieces together here: https://github.com/YairHalberstadt/stronginject/pull/143#discussion_r687033329